packages/clay-link/src/index.tsx

```diff
@@ -45,15 +57,34 @@ const ClayLink = React.forwardRef<HTMLAnchorElement, IProps>(
 	) => {
 		const TagOrComponent = React.useContext(ClayLinkContext);
 
+		let classes;
+
+		if (button) {
+			button = button === true ? {} : button;
+
+			classes = {
+				btn: !!button,
+				'btn-block': button?.block,
```

Comment:
It's a shame you need the `?.` here, because I would have thought that TS would have been able to narrow the type; eg.

- At top, we know `button` is `boolean`, `{}` or `undefined`.
- After `if`, we know it is `true` or `{}`.
- After `button =` reassignment, we know it is `{}`

It seems that if you assign it to a variable then it *can* narrow it for sure (eg. [playground](https://www.typescriptlang.org/play/index.html#code/GYVwdgxgLglg9mABADwBQChFcQIxFKBAfgC5NsKAfXOOAGwFMBDMciragbzfd5zrgQA1qRr1mYADQ9eFALYI4AZwAOTCAwAmonLUYtps2UrlM6dHXomGjiAL7oAlIm4UYwRKjwEEz17IgEJShEMCYAJ3C4AHctRABeXHxCJHi0xChwkAZEIhc7RBIknzAAbh4HByA)):

```js
function x(
    button?:
        | boolean
        | {
              block?: boolean,
              monospaced?: boolean,
              small?: boolean,
          }
) {
    if (button) {
        /*
         * Type of `narrowed` inferred here is:
         *
         *     {
         *         block?: boolean | undefined;
         *         monospaced?: boolean | undefined;
         *         small?: boolean | undefined;
         *     }
         */
        const narrowed = button === true ? {} : button;
    }
}
```

Reply:
@wincent yeah I think I initially tried to use the optional chaining and was still getting errors, so then I added the assignment and it was fine. But you are right, the chaining was redundant and not needed, I just removed it in the latest commit.